Add security-council doc and updated 2 docs (#75)
maticnetwork/zkevm-docs: docs/protocol/admin-role.md

  - timelock controller
---

> Polygon zkEVM's _Admin_ is consists of three (3) developers of the Polygon team, who oversee any upgrades of the zk-rollup software.

Reviewer: ```suggestion
Polygon zkEVM's _Admin_ is consists of three (3) developers of the Polygon team, who oversee any upgrades of the zk-rollup software.
```
```suggestion
Polygon zkEVM's _Admin_ consists of three (3) developers of the Polygon team, who oversee any upgrades of the zk-Rollup software.
```

> Polygon zkEVM's _Admin_ is consists of three (3) developers of the Polygon team, who oversee any upgrades of the zk-rollup software.

Reviewer: ```suggestion
Polygon zkEVM's _Admin_ consists of three (3) developers of the Polygon team, who oversee any upgrades of the zk-Rollup software.
```

> Whenever there are bug-fixes or updates to be made, the Admin uses a special _Admin Multisig Contract_ to approve the changes.

Reviewer: ```suggestion
Whenever there are bug fixes or updates to be made, the Admin uses a special _Admin Multisig Contract_ to approve the changes.
```

Approval by any two of the three Admin members (_2-out-of-3_) is required.

> However, there's minimum 10-day waiting period before the changes get executed.

Reviewer: ```suggestion
However, there's a minimum 10-day waiting period before the changes get executed.
```

The 10-day delay allows users to carefully assess the proposed changes and decide whether to exit or not.

> Another smart contract called _Timelock Contract_ is responsible for enabling the 10-day delay.

Reviewer: ```suggestion
Another smart contract called the _Timelock Contract_ is responsible for enabling the 10-day delay.
```

> An outline of the _upgrade process_ is outlined [here](/docs/protocol/zkevm-upgrades-process.md#process-overview).

Reviewer: ```suggestion
An outline of the _upgrade process_ is outlined [here](/protocol/zkevm-upgrades-process.md#process-overview).
```

:::

The **Timelock Controller** has been added to the zkEVM Protocol in order to improve user security and confidence.

The Admin can schedule and commit maintenance operations transactions in L1 using a Timelock Controller, and the timelock can be activated to carry out the transactions when a specified `minDelay` time has passed.

**The Polygon zkEVM team has decided to use the [OpenZeppelin's TimelockController.sol contract](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/TimelockController.sol) to inherit security as well to avoid the lengthy and complicated audit process**. We have changed the `getMinDelay` method in the contract and this modified implementation is named **PolygonZkEVMTimelock.sol** contract.

In the event that the zkEVM contract system is in [Emergency Mode](emergency-state.md), the new `getMinDelay` will set the time `minDelay` to 0. This is when the Security Council Multisig takes control.

The zKEVM Protocol’s Admin role is set to an instance of **PolygonZkEVMTimelock.sol** contract address since the deployment of the zk-rollup.

## Governance of zKEVM Contracts

> The Admin carries significant and crititcal responsibility and that's the reason why it is coomposed of three (3) members as opposed to just one person. For this reason, the Admin Ethereum account of a PolygonZkEVMTimelock.sol contract instance is assigned to a multisig contract that acts as a governance tool for the zkEVM Protocol, therefore achieving some degree of decentralization overall.

Reviewer: ```suggestion
The Admin carries significant and critical responsibility, which is why it is composed of three (3) members as opposed to just one person. For this reason, the Admin Ethereum account of a `PolygonZkEVMTimelock.sol` contract instance is assigned to a multisig contract that acts as a governance tool for the zkEVM Protocol, therefore achieving some degree of overall decentralization.
```
